pkg/api/pod/warnings.go

```diff
@@ -167,10 +167,25 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 		}
 	}
 
-	// removed volume plugins
+	// deprecated and removed volume plugins
 	for i, v := range podSpec.Volumes {
 		if v.PhotonPersistentDisk != nil {
-			warnings = append(warnings, fmt.Sprintf("%s: non-functional in v1.16+", fieldPath.Child("spec", "volumes").Index(i).Child("photonPersistentDisk")))
+			warnings = append(warnings, fmt.Sprintf("%s: deprecated in v1.11, non-functional in v1.16+", fieldPath.Child("spec", "volumes").Index(i).Child("photonPersistentDisk")))
+		}
+		if v.GitRepo != nil {
+			warnings = append(warnings, fmt.Sprintf("%s: deprecated in v1.11", fieldPath.Child("spec", "volumes").Index(i).Child("gitRepo")))
+		}
+		if v.ScaleIO != nil {
+			warnings = append(warnings, fmt.Sprintf("%s: deprecated in v1.16, non-functional in v1.22+", fieldPath.Child("spec", "volumes").Index(i).Child("scaleIO")))
+		}
+		if v.Flocker != nil {
+			warnings = append(warnings, fmt.Sprintf("%s: deprecated in v1.22, support removal is planned in v1.24", fieldPath.Child("spec", "volumes").Index(i).Child("flocker")))
```

Comment:
To meet the 1 year feature deprecation policy, we cannot remove it until 1.26 (assuming 3 releases a year)

Reply:
Done. I guess we might want to deprecate csi migration enabled plugins earlier then.

Reply:
The difference is that the api will still work with migrated plugins